test/CodeGen/AVR/smul-with-overflow.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: llc < %s -march=avr | FileCheck %s
+
+define i1 @signed_multiplication_did_overflow(i8, i8) unnamed_addr {
+; CHECK-LABEL: signed_multiplication_did_overflow:
```

Comment:
```asm
signed_multiplication_did_overflow:     ; @signed_multiplication_did_overflow
; BB#0:                                 ; %entry-block
	muls	r24, r22
	mov	r25, r1
	eor	r1, r1
	mov	r18, r0
	asr	r18
	asr	r18
	asr	r18
	asr	r18
	asr	r18
	asr	r18
	asr	r18
	ldi	r24, 1
	cp	r25, r18
	brne	LBB0_2
; BB#1:                                 ; %entry-block
	ldi	r24, 0
LBB0_2:                                 ; %entry-block
	ret
```

This is a bit more involved than unsigned. What shape of tests would you like to see here?